File: app/src/main/java/club/gitmad/chat/MessageAdapter.java
package club.gitmad.chat;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class MessageAdapter extends RecyclerView.Adapter<MessageAdapter.ViewHolder> {
    private List<Message> messages;

    static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvMessage;
        TextView tvSender;

        ViewHolder(View view) {
            super(view);
            this.tvMessage = view.findViewById(R.id.tvMessage);
            this.tvSender = view.findViewById(R.id.tvSender);
        }

        private void setMessage(Message message) {
            tvMessage.setText(message.getMessage());
            tvSender.setText(message.getSender());
        }

    }

    MessageAdapter(List<Message> messages) {
        this.messages = messages;
    }

    @NonNull
    @Override
    public MessageAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.row_message, parent, false));
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.setMessage(messages.get(position));
    }

    @Override
    public int getItemCount() {
        return messages.size();
    }
}

File: app/src/main/java/club/gitmad/chat/MessageAdapter.kt
package club.gitmad.chat

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.row_message.view.*

class MessageAdapter(private val messages: MutableList<Message>) : RecyclerView.Adapter<MessageAdapter.ViewHolder>() {

    class ViewHolder(override val containerView: View) : RecyclerView.ViewHolder(containerView), LayoutContainer {
        fun bindMessage(message: Message) {
            containerView.tvMessage.text = message.message
            containerView.tvSender.text = message.sender
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.row_message, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bindMessage(messages[position])
    }

    override fun getItemCount() = messages.size
}
